tests/pthread/hello_thread.c

```diff
@@ -12,15 +12,14 @@
 void *thread_main(void *arg)
 {
 	EM_ASM(out('hello from thread!'));
-#ifdef REPORT_RESULT
-	REPORT_RESULT(1);
-#endif
-	return 0;
+	emscripten_force_exit(0);
+	__builtin_unreachable();
```

Comment:
are the unreachables necessary? Would adding the `noreturn` attribute in the header achieve the same?

Reply:
I guess `noreturn` might work, but this is more explicit and will for sure trap if `emscripten_force_exit` was ever to return for some reason.

I like adding it here so it clear the reader that we don't return any value from this function.

Reply:
Adding `noreturn` would help all the places that call this method, though? That would help users as well, as this isn't an internal API.

Anyhow, that can be separate from this PR.

Reply:
Yes, clearly emscripten_force_exit should be noreturn.. I hadn't realized that it wasn't.  But I think there is also value in adding unreachable here for test readability.